.gitignore

```diff
@@ -398,3 +398,6 @@ FodyWeavers.xsd
 
 # JetBrains Rider
 *.sln.iml
+
+# Current project custom files
+Source/BSN.IpTables.Cli/generated

```

Comment:
I do not thing it is good, because all custom module and docs and other things is ignroe too

Reply:
Currently we see uncommitted generated code in our source when working which is annoying in git commands.

Generally speaking, there are three solutions:

1. Ignoring the generated folder completely.
2. Committing all files in the generated folder.
3. Committing changed/added files in the generated folder, and ignoring others.

I disagree with solution 2, because I think having generated code in our source is not a good idea. It contains countless files and lines, it may change unintentionally. Maintaining it is an absurd work.

About solution 3, I ran a test. I created a file: "generated\custom\Custom.ps1", then ran Autorest to see if it remains there or not (for example may Autorest remove the generated folder at first or not). The result was that the file remains there. So this solution may work, but the gitignore command became confusing because we should ignore an entire folder except for some files in it.

Another solution that comes to mind is to commit custom commands somewhere else and then copy them into the generated folder in the build process.

Please comment on these solutions.

Reply:
Suggestion: Choosing an easy solution (ignoring all or do nothing) for now, and delay actual decision to next commit when custom commands are going to be added actually.

Reply:
Ok